src/python/pants/engine/internals/graph_test.py

```diff
@@ -73,6 +73,7 @@
 from pants.engine.unions import UnionMembership, UnionRule, union
```

Comment:
We use a couple distinct `RuleRunner`s here, more than we had `TestBase`s. This results in tighter graphs.